ahk/templates/daemon.ahk

```diff
@@ -518,6 +518,14 @@ AHKWinWait(ByRef command) {
     timeout := command[4]
     extitle := command[5]
     extext := command[6]
+    match_mode:= command[7]
+    match_speed := command[8]
+    if (match_mode != "None") {
+        SetTitleMatchMode %match_mode%
+    }
+    if (match_speed != "None") {
+        SetTitleMatchMode %match_speed%
+    }
     WinWait,%title%,%text%,%timeout%,%extitle%,%extext%
```

Comment:
One thing that's important in the daemon is that changes made with `SetTitleMatchMode` will persist even after the function is done. This may result in unexpected behavior for subsequent method calls that are affected by `SetTitleMatchMode`

for example:

```python
ahk.win_wait(..., match_mode=3)
ahk.win_get(...)  # unexpectedly uses `SetTitleMatchMode, 3` behavior
```

Hence, we probably want to record the original value before invoking WinWait command and then revert to the original value after invoking the command.

```suggestion
    previous_match_mode := %A_TitleMatchMode%
    previous_match_speed := %A_TitleMatchModeSpeed%
    if (match_mode != "None") {
        SetTitleMatchMode %match_mode%
    }
    if (match_speed != "None") {
        SetTitleMatchMode %match_speed%
    }
    WinWait,%title%,%text%,%timeout%,%extitle%,%extext%
    ; Revert to original match modes
    SetTitleMatchMode, %previous_match_mode%
    SetTitleMatchMode, %previous_match_speed%
```

(the above suggestion is untested, but I think that's the right idea)

We should probably also add a test for this.

Reply:
Thank you for the suggestion.  :)
If there are any problems, please point them out.
[b2ea856](https://github.com/spyoungtech/ahk/pull/167/commits/b2ea85634fbd68c74f366ba0cae0f93c5c51a955)